libs/application-generic/src/usecases/create-workflow/create-workflow.command.ts

```diff
@@ -38,38 +39,47 @@ import { Type } from 'class-transformer';
 import { EnvironmentWithUserCommand } from '../../commands';
 import { PreferencesRequired } from '../upsert-preferences';
 
+export const MAX_TAG_ELEMENTS = 16;
+export const MAX_TAG_LENGTH = 32;
+export const MAX_NAME_LENGTH = 64;
+export const MAX_DESCRIPTION_LENGTH = 256;
+
 export class CreateWorkflowCommand extends EnvironmentWithUserCommand {
-  @IsMongoId()
   @IsDefined()
-  notificationGroupId?: string;
+  @IsString()
+  @Length(1, MAX_NAME_LENGTH)
+  name: string;
 
+  @IsString()
   @IsOptional()
-  notificationGroup?: INotificationGroup;
+  @Length(0, MAX_DESCRIPTION_LENGTH)
+  description?: string;
 
   @IsOptional()
   @IsArray()
+  @ArrayUnique()
+  @Length(1, MAX_TAG_LENGTH, { each: true })
   tags?: string[];
```

Comment:
`class-validators` wasn't working with `ArrayMaxSize`, so I've removed it for now. I will revisit this validation in a follow-up PR